helix-core/src/indent.rs

```diff
@@ -73,9 +73,12 @@ pub fn auto_detect_indent_style(document_text: &Rope) -> Option<IndentStyle> {
     // Build a histogram of the indentation *increases* between
     // subsequent lines, ignoring lines that are all whitespace.
     //
-    // Index 0 is for tabs, the rest are 1-MAX_INDENT spaces.
-    let histogram: [usize; MAX_INDENT as usize + 1] = {
-        let mut histogram = [0; MAX_INDENT as usize + 1];
+    // Index 1 is for tabs, the rest are 1-MAX_INDENT spaces.
```

Comment:
This comment should now also explain what index 0 is.  Something like "Index 0 is a count of lines with no leading white space, ...".